### PR TITLE
Add conversion from Zarr array to tensorstore

### PR DESCRIPTION
**Pull request by ilan-francisco**
Enabling tensorstore conversion from a zarr array, as mentioned in #26.

### CODE REVIEW

**Comment by JoOkuma** [comment]
Thanks for the PR, @ilan-francisco.

I have a few minor comments:
- Could you add a test and document the method?
- You also need to add the package as a dependency. If it's an optional dependency, `pytest` can skip the test if the import fails.
- Convert the `assert` to an `if` and `raise` call.

@ziw-liu , will `tensorstore` be an optional dependency?

**Comment by ziw-liu** [comment]
> will `tensorstore` be an optional dependency?

This sounds like the best solution as the complied extension of tensorstore is not available on every platform.

If we do go this route then this PR (and future ones using ts) will need to handle import errors.

File: iohub/ngff.py [hunk 1]
```diff
@@ -340,7 +341,19 @@ def downscale(self):
         raise NotImplementedError
 
     def tensorstore(self):
-        raise NotImplementedError
+        import tensorstore as ts
+
+        assert self.path != "", "Zarr array must be stored on a file system."
```
Comment (ziw-liu) [inline iohub/ngff.py:346]: In theory an iohub NGFF store is always on a local FS, so this might be redundant.

**Comment by ziw-liu** [comment]
@JoOkuma Is further follow-up work expected for this PR?

**Comment by ziw-liu** [comment]
Closing as stale.

**Pull request closed without merging**